### PR TITLE
feat: website with open api docs

### LINKED ISSUE #69: SuperHot OpenAPI docs

### PR DESCRIPTION
**Pull request by vasco-santos**
Closes #69 

You can see preview at https://bea67338.nftstorage-link.pages.dev/api-docs

### CODE REVIEW

File: packages/website/public/schema.yml [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+  - url: http://127.0.0.1:8787
+
+tags:
+  - name: nftstorage.link
+paths:
+  /perma-cache/{url}:
+    post:
+      tags:
+        - nftstorage.link
+      summary: Put a given URL response in nftstorage.link perma-cache.
```
Comment (alanshaw) [inline packages/website/public/schema.yml:17]: I would avoid using the word "Put" for this HTTP POST route.

```suggestion
      summary: Add the content of the given URL to the perma-cache.
```

File: packages/website/public/schema.yml [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+      description: |
+        Store the response of the nftstorage.link IPFS gateway in a edge perma-cache
```
Comment (alanshaw) [inline packages/website/public/schema.yml:19]:
```suggestion
        Store the response of the nftstorage.link IPFS gateway in an edge perma-cache
```

File: packages/website/public/schema.yml [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+        to accelerate future request to the gateway.
+
+        Note: During beta this is limited to 4.995 GB of content length per URL.
```
Comment (alanshaw) [inline packages/website/public/schema.yml:22]:
```suggestion
        Note: During beta this is limited to 4.995 GB of content per URL.
```

File: packages/website/public/schema.yml [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+  - name: nftstorage.link
+paths:
+  /perma-cache/{url}:
+    post:
+      tags:
+        - nftstorage.link
+      summary: Put a given URL response in nftstorage.link perma-cache.
+      description: |
+        Store the response of the nftstorage.link IPFS gateway in a edge perma-cache
+        to accelerate future request to the gateway.
```
Comment (alanshaw) [inline packages/website/public/schema.yml:20]:
```suggestion
        to accelerate future requests to the nftstorage.link gateway.
```

File: packages/website/public/schema.yml [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+        '403':
+          $ref: '#/components/responses/forbidden'
+        '409':
+          $ref: '#/components/responses/conflict'
+        '500':
+          $ref: '#/components/responses/internalServerError'
+    delete:
+      tags:
+        - nftstorage.link
+      summary: Delete a given URL response from nftstorage.link perma-cache.
```
Comment (alanshaw) [inline packages/website/public/schema.yml:54]:
```suggestion
      summary: Remove content for the given URL from the perma-cache.
```

File: packages/website/public/schema.yml [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+      description: |
+        Delete a given URL response from nftstorage.link perma cache.
```
Comment (alanshaw) [inline packages/website/public/schema.yml:56]:
```suggestion
        Remove previously added content for the given URL from the perma-cache.
```

File: packages/website/public/schema.yml [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+          $ref: '#/components/responses/forbidden'
+        '404':
+          $ref: '#/components/responses/notFound'
+        '500':
+          $ref: '#/components/responses/internalServerError'
+  /perma-cache:
+    get:
+      tags:
+        - nftstorage.link
+      summary: Get all URLs perma cached in nftstorage.link by the user.
```
Comment (alanshaw) [inline packages/website/public/schema.yml:91]:
```suggestion
      summary: List the URLs that have been added to the perma-cache.
```

File: packages/website/public/schema.yml [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+      description: |
+        Get all URLs previously perma-cached by the user.
```
Comment (alanshaw) [inline packages/website/public/schema.yml:93]:
```suggestion
        Retrieve a paginated list of the URLs that have been added to the perma-cache.
```

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: packages/website/public/schema.yml [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+          default: true
+        value:
+          $ref: '#/components/schemas/PermaCacheEntry'
+    PermaCacheGetResponse:
+      type: object
+      properties:
+        ok:
+          type: boolean
+          default: true
+        value:
```
[the remark this answers is not in the document]
Comment (vasco-santos) [reply packages/website/public/schema.yml]: Good catch thanks, fixed now

**Review by alanshaw — APPROVED** [review]
LGTM but please fix the typo!

**Pull request merged**